src/js/client/renderer.js

```diff
@@ -97,7 +98,13 @@
             }
         },
         events: {
-            "onTemplatesLoaded": null
+            "onTemplatesLoaded": null,
+            "onAllResourcesLoaded": {
```

Comment:
If we decide to use async activities to populate templates and messages, one option would be to have a promise chained event that handles the loading and which fires `onAllResourcesLoaded` as its last step.